design/one-pager-k8s-native-providers.md

```diff
@@ -0,0 +1,562 @@
+kind: PostgreSQLInstanceClass
+metadata:
+  name: postgresql-standard
+  namespace: app-project1-dev
+classRef:
+  kind: ClusterClass
+  apiVersion: cockroachdb.rook.crossplane.io/v1alpha1
+  name: rook-cockroachdb
+  namespace: rook-infra-dev
+```
+
+4. Create a `PostgreSQLInstance` claim:
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  name: app-postgresql
+  namespace: app-project1-dev
+spec:
+  classRef:
+    name: postgresql-standard
+  writeConnectionSecretToRef:
+    name: postgresqlconn
+  engineVersion: "9.6"
+```
+
+On creation of the `PostgreSQLInstance` claim, the claim controller would
+configure and create a `Cluster` (`cockroachdb.rook.crossplane.io/v1alpha1`)
+instance. Then, the managed controller would use the Crossplane `Cluster`
+instance to create a `Cluster` (`cockroachdb.rook.io/v1alpha1`) in the target
+Kubernetes cluster where the Rook CockroachDB operator is installed. This
+workflow is similar to dynamic provisioning in a typical infrastructure
+provider:
+
+![K8s Native Providers](./images/k8s-native-providers.png)
+
+### Non-Portable Resources
+
+Some resources exposed by Kubernetes-native infrastructure providers may not be
+initially supported by dynamic provisioning. This is true for existing
+infrastructure stacks as well, as types like `VirtualNetwork` in Azure can be
+provisioned statically, but not in the dynamic `claim <-> class` pattern. These
+types of resources may eventually be supported by portable claim types in core
+Crossplane.
+
+An example of this type of resource for Rook would be a [Minio] `ObjectStore`.
+While Crossplane does support a `Bucket` claim type for dynamic provisioning of
+object storage, the Minio `ObjectStore` type is actually setting up a storage
+service, much like AWS S3. Once an `ObjectStore` is provisioned, individual
+object storage "buckets" can be created. However, Rook does not currently expose
+a Kubernetes custom resource kind to provision buckets in the Minio object
+store, so buckets must be created directly by interacting with the Minio API.
+
+For an initial implementation of the Rook stack, users should be able to
+provision Minio `ObjectStores` statically, but support for creating buckets in
+that object store will likely either be deferred to a later implementation (see
+[Future Considerations]). 
+
+It may also be desirable to create an `ObjectStore` claim type in Crossplane in
+order to be able dynamically provision `ObjectStores` in Rook. However, like the
+networking types in other stacks, it is likely not a pressing issue to support
+dynamic provisioning for this resource type in the initial implementation.
+
+## Technical Design
+
+### The Kubernetes Provider Kind
+
+The Kubernetes `Provider` kind will be implemented in core Crossplane in a new
+API group: `kubernetes.crossplane.io`. This will allow it to be used by core
+Crossplane custom resources as well as Kubernetes-native infrastructure stacks
+like Rook.
```

Comment:
I think this makes sense for now, but I wonder if one day the Kubernetes provider and workloads move into `stack-kubernetes`. 🤔

Reply:
I actually would be in favor of keeping it in core Crossplane as I see it as the appropriate place for resources that may be utilized my multiple stacks.